numpy/core/src/common/simd/neon/math.h

```diff
@@ -153,4 +153,20 @@ NPY_FINLINE npyv_s64 npyv_min_s64(npyv_s64 a, npyv_s64 b)
     return vbslq_s64(npyv_cmplt_s64(a, b), a, b);
 }
 
+// ceil
+#ifdef NPY_HAVE_ASIMD
+    #define npyv_ceil_f32 vrndpq_f32
+#else
+   NPY_FINLINE npyv_f32 npyv_ceil_f32(npyv_f32 a)
+   {
+       npyv_f32 conv_trunc = vcvtq_f32_s32(vcvtq_s32_f32(a));
+       npyv_f32 conv_trunc_add_one = npyv_add_f32(conv_trunc, vdupq_n_f32(1.0f));
+       npyv_u32 mask = vcltq_f32(conv_trunc, a);
+       return vbslq_f32(mask, conv_trunc, conv_trunc_add_one);
+   }
```

Comment:
```suggestion
NPY_FINLINE npyv_f32 npyv_ceil_f32(npyv_f32 a)
    {
        const npyv_s32 szero = vreinterpretq_s32_f32(vdupq_n_f32(-0.0f));
        const npyv_u32 one = vreinterpretq_u32_f32(vdupq_n_f32(1.0f));
        const npyv_s32 max_int = vdupq_n_s32(0x7fffffff);
        /**
         * On armv7, vcvtq.f32 handles special cases as follows:
         *  NaN return 0
         * +inf or +outrange return 0x80000000(-0.0f)
         * -inf or -outrange return 0x7fffffff(nan)
         */
        npyv_s32 roundi = vcvtq_s32_f32(a);
        npyv_f32 round = vcvtq_f32_s32(roundi);
        npyv_f32 ceil = vaddq_f32(round, vreinterpretq_f32_u32(
            vandq_u32(vcltq_f32(round, a), one))
        ); 
        // respect signed zero, e.g. -0.5 -> -0.0
        npyv_f32 rzero = vreinterpretq_f32_s32(vorrq_s32(
            vreinterpretq_s32_f32(ceil),
            vandq_s32(vreinterpretq_s32_f32(a), szero)
        ));
        // if nan or overflow return a
        npyv_u32 nnan = npyv_notnan_f32(a);
        npyv_u32 overflow = vorrq_u32(
            vceqq_s32(roundi, szero), vceqq_s32(roundi, max_int)
        );
        return vbslq_f32(vbicq_u32(nnan, overflow), rzero, a);
    }

```

Similar to the other architectures, ceil_f32 should respect special cases and signed zero.
My suggestion here is similar to the SSE version except for handling finite and outranges due to the nature of `vcvtq_s32_f32`.

Note that CI doesn't cover armv7/neon neither aarch32/ASIMD so I had to test it on my local machine and it seems good to me.

Reply:
update: we have now a CI/workflow for testing universal intrinsics on armv7/aarch32.